### PR TITLE
Add new snapshot tests for overlapping delete in shared matrix

### PR DESCRIPTION
**Pull request by anthony-murphy**
Test Data Change: https://github.com/microsoft/FluidFrameworkTestData/pull/38

- Adds new test data
- some minor changes in how we report errors to also print the file name.
- generate new file snapshots where the files has snapshots
-  suppress error

related #6957 #9138

### CODE REVIEW

File: packages/tools/replay-tool/src/replayMessages.ts [hunk 1]
```diff
@@ -349,14 +349,17 @@ export class ReplayTool {
 
         // GC will consider unreferenced node in old documents to be inactive. When such nodes receive ops or are
         // loaded, GC will log an error. Ignore those errors since we don't care about them when replaying old docs.
-        if (event.eventName.includes("GarbageCollector:inactiveObject_")) {
+        if (event.eventName.includes("GarbageCollector:inactiveObject_")
+            // there is something wrong with summary ops, but we don't run the summarizer, so it shouldn't matter
+            || event.eventName.includes("SummaryAckWithoutOp")
```
Comment (anthony-murphy) [inline packages/tools/replay-tool/src/replayMessages.ts:354]: SummaryAckWithoutOp being logged when running SharedMatrixDeletes snapshot tests
we should figure out why this is being logged, fix it, and remove this supression

File: packages/test/snapshots/src/replayMultipleFiles.ts [hunk 2]
```diff
@@ -103,8 +103,13 @@ export async function processOneNode(args: IWorkerArgs) {
     replayArgs.inDirName = args.folder;
     // The output snapshots to compare against are under "currentSnapshots" sub-directory.
     replayArgs.outDirName = `${args.folder}/${currentSnapshots}`;
-    replayArgs.snapFreq = args.snapFreq;
-
+    if (args.mode === Mode.NewSnapshots) {
+        // when generating new snapshots, match those from
+        // the original file based on summarize ops
+        replayArgs.testSummaries = true;
```
Comment (agarwal-navin) [inline packages/test/snapshots/src/replayMultipleFiles.ts:109]: What does this do? I am curious for my own understanding of the tests and replay tool.

File: packages/test/snapshots/src/replayMultipleFiles.ts [hunk 3]
```diff
@@ -216,7 +220,7 @@ async function processNodeForValidate(
 }
 
 /**
- * In UdpateSnapshots mode, the snapshot format has changed and we need to update the reference snapshot files with the
+ * In UpdateSnapshots mode, the snapshot format has changed and we need to update the reference snapshot files with the
```
Comment (agarwal-navin) [inline packages/test/snapshots/src/replayMultipleFiles.ts:223]: I need a spell checker!!

File: packages/tools/replay-tool/src/replayMessages.ts [hunk 4]
```diff
@@ -514,20 +516,16 @@ export class ReplayTool {
 
     private async mainCycle() {
         const originalSummaries =
-            this.args.testSummaries ? [...this.mainDocument.originalSummarySequenceNumbers] : [];
-        let nextSnapPoint;
-        do {
-            nextSnapPoint = originalSummaries.shift() ?? this.args.from;
-        } while (nextSnapPoint < this.args.from);
+            this.args.testSummaries
```
Comment (agarwal-navin) [inline packages/tools/replay-tool/src/replayMessages.ts:519]: Is this a change in behavior or just code restructuring?

File: packages/tools/replay-tool/src/replayMessages.ts (hunk 1 above)
Comment (agarwal-navin) [reply packages/tools/replay-tool/src/replayMessages.ts:354]: I repro'd this locally and I think I know what is happening. The snapshot frequency for the stress test is 50 which means that after every 50 ops, it take a snapshot of the container, loads a new container from this snapshot and replays rest of the ops (among other things).
In the `messages.json` file that you added, op# 100 is `summarize` and op# 101 is `summaryAck`. So, at some point, a container loads from snapshot at op# 100 and sees a `summaryAck`. It thinks it got a summary ack without an op. If the snap point was an op above or below, you would not see this errror.

File: packages/tools/replay-tool/src/replayMessages.ts (hunk 4 above)
Comment (anthony-murphy) [reply packages/tools/replay-tool/src/replayMessages.ts:519]: just code restructuring. i actually wrote this code long ago and it was so bad i had to make it better :)

Comment (agarwal-navin) [reply packages/tools/replay-tool/src/replayMessages.ts:519]: Haha okay. I get the same feeling with some of my code :-).

**Review by agarwal-navin — APPROVED** [review]

**Pull request merged**